jcl/src/java.base/share/classes/java/lang/invoke/ArrayVarHandle.java

```diff
@@ -143,6 +126,10 @@ static final void boundsCheck(int arrayLength, long index) {
 			}
 		}
 		
+		static final long computeIndex(int index, int indexOffset, int indexScale) {
+			return indexOffset + ((long)index * indexScale);
+		}
```

Comment:
The method that this replaces was poorly named: I suggest 'computeOffset' would be better. Also the second argument would be better named 'baseOffset' to reflect the origin of that value.

Reply:
Thanks! Will fix it.

Reply:
Fixed